src/test/java/com/exasol/adapter/request/parser/PushDownSqlParserTest.java

```diff
@@ -566,6 +566,52 @@ void testParseFunctionScalarExtract() {
                 () -> assertThat(sqlLiteralTimestamp.getValue(), equalTo("2015-12-01 12:01:01.1234")));
     }
 
+    @Test
+    void testParseFunctionScalarCastToVarchar() {
+        final String sqlAsJson = "{" //
+                + "   \"type\" : \"function_scalar_cast\", " //
+                + "   \"name\" : \"CAST\", " //
+                + "   \"dataType\" : { " //
+                + "        \"type\" : \"VARCHAR\", " //
+                + "        \"size\" : 10000 " //
+                + "   }, " //
+                + "   \"arguments\" : [ " //
+                + "   { " //
+                + "        \"type\" : \"literal_double\", " //
+                + "        \"value\" : \"1.234\" " //
+                + "   } " //
+                + "   ] " //
+                + "}";
+        final JsonObject jsonObject = createJsonObjectFromString(sqlAsJson);
+        final SqlFunctionScalarCast sqlFunctionScalarCast = (SqlFunctionScalarCast) this.defaultParser
+                .parseExpression(jsonObject);
+        final SqlLiteralDouble sqlLiteralDouble = (SqlLiteralDouble) sqlFunctionScalarCast.getArgument();
+        assertAll(() -> assertThat(sqlFunctionScalarCast.getType(), equalTo(FUNCTION_SCALAR_CAST)),
+                () -> assertThat(sqlLiteralDouble.getValue(), equalTo(1.234)));
```

Comment:
This test contains two blind casts. I know what you are testing here, but this somehow feels wrong. I cannot offer a more compact solution though, I you did not do the cast, the test would get more verbose, which might make it even harder to read.